utilities/deep_inspection.py

```
@@ -334,28 +363,65 @@ def verify_di_results(conversion: Conversion, vm_name: str) -> None:
+    Args:
+        plan_resource: The Plan CR to check.
+
+    Raises:
+        AssertionError: If any DeepInspection CRs are found.
+    """
+    conversions = get_plan_conversion_crs(plan_resource=plan_resource, conversion_type="DeepInspection")
+    assert not conversions, (
+        f"Plan '{plan_resource.name}': Expected no DeepInspection CRs (run_preflight_inspection=False), "
+        f"but found {len(conversions)}: {[c.name for c in conversions]}."
     )
+    LOGGER.info(f"Plan '{plan_resource.name}': Confirmed no DeepInspection CRs exist (DI correctly skipped).")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)